jmaqs-selenium/src/test/java/com/magenic/jmaqs/selenium/BaseSeleniumPageModelUnitTest.java

```diff
@@ -0,0 +1,99 @@
+import com.magenic.jmaqs.utilities.helper.exceptions.TimeoutException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class BaseSeleniumPageModelUnitTest extends BaseSeleniumTest {
+
+  @Test
+  public void testGetLogger() {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    Assert.assertNotNull(pageElementsPageModel.getLogger());
+  }
+
+  @Test
+  public void testGetTestObject() {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    Assert.assertNotNull(pageElementsPageModel.getTestObject());
+  }
+
+  @Test
+  public void testGetWebDriver() {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    Assert.assertNotNull(pageElementsPageModel.getWebDriver());
+  }
+
+  @Test
+  public void testGetPerfTimerCollection() {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    Assert.assertNotNull(pageElementsPageModel.getPerfTimerCollection());
+  }
+
+  @Test
+  public void testSetWebDriver() throws Exception {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    int hashCode = pageElementsPageModel.getWebDriver().hashCode();
+    pageElementsPageModel.setWebDriver(this.getBrowser());
+    int hashCode1 = pageElementsPageModel.getWebDriver().hashCode();
+    Assert.assertNotEquals(hashCode, hashCode1);
+  }
+
+  @Test
+  public void testIsPageLoaded() {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    pageElementsPageModel.open();
+    Assert.assertTrue(pageElementsPageModel.isPageLoaded());
+  }
+
+  @Test
+  public void testGetLazyElement() throws TimeoutException, InterruptedException {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    pageElementsPageModel.open();
+    pageElementsPageModel.waitForPageLoad();
+    LazyWebElement lazyElement = pageElementsPageModel.getLazyElement(pageElementsPageModel.pageTitleLocator);
+    Assert.assertNotNull(lazyElement);
+    Assert.assertEquals(lazyElement.getText(), "Elements to be automated");
+
+  }
+
+  @Test
+  public void testGetLazyElement1() throws TimeoutException, InterruptedException {
+    PageElementsPageModel pageElementsPageModel = new PageElementsPageModel(getTestObject());
+    pageElementsPageModel.open();
+    pageElementsPageModel.waitForPageLoad();
+    LazyWebElement lazyElement = pageElementsPageModel
+        .getLazyElement(pageElementsPageModel.pageTitleLocator, "Page Title");
+    Assert.assertNotNull(lazyElement);
+    Assert.assertEquals(lazyElement.getText(), "Elements to be automated");
+    Assert.assertEquals(lazyElement.getUserFriendlyName(), "Page Title");
+  }
+
+  @Test
+  public void testGetLazyElement2() throws TimeoutException, InterruptedException {
```

Comment:
We should rename these methods to be more specific. for example, since this method tests calling getLazyElement() using the by and parent, this test could be renamed testGetLazyElementWithParentElementAndBy()

as well,  I think we are lacking validating if the lazyElementMap is set. Currently we know that the getElementMethod returns back a lazy element, but perhaps we can add the following:
1. using getLazyElement to create a new element, then call getLazyElement again and verify the reference is the same
2. Call create lazy Element and verify the element is actually stored in the map (this would require adding a getter for the lazyElementStore). Perhaps do this for each override